### PR TITLE
Add docker

### PR DESCRIPTION
**Pull request by quvide**
## Description
This PR adds Dockerfiles for local development and a production environment.

### Local development
* Directory: `docker/development`

In this configuration static files and the dynamic api are served by an Express server running inside a Docker container. The files are built during the construction of the container, not while it's running. This means the container must be built again after making changes.

### Production
* Directory: `docker/production`

In this configuration there are two Docker containers. One runs nginx, which passes api requests to the backend container and serves other requests from the built files. The containers must be built again after making changes.

## Future work
* It would benefit developers to not require building containers again after making changes, so at least the development setup should watch for changes and automatically build new versions. If this isn't possible, it should at least mount files from the host system so that the containers should only be restarted

* Could the development and production environments be somehow merged to ease and simplify maintenance when the application changes?

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
     "lint": "eslint .",
-    "build": "webpack -p --mode production --config webpack.production.config.js",
+    "build": "sh build.sh",
```
Comment (quvide) [inline client/package.json]: Is there a way to do this in webpack or some automation tool?

Comment (quvide) [reply client/package.json]: > I have been avoiding shell scripts in npm scripts as they may not work on Windows!

Should we add an automation tool like gulp or grunt?

**Comment by quvide** [comment]
> I have been avoiding shell scripts in npm scripts as they may not work on Windows!

Should we add an automation tool like gulp or grunt?

EDIT: This was resolved by just using webpack for it

**Comment by HenrikHelenius** [comment]
[[epic] [dev prep] Prepare dev related stuff Henrik](https://trello.com/c/9TrGVUQE/28-epic-dev-prep-prepare-dev-related-stuff-henrik)

**Review by bqqbarbhg — COMMENTED** [review]
Ship it

**Review by bqqbarbhg — APPROVED** [review]
Ship it for real

**Pull request merged**